gnomad/resources/resource_utils.py

```diff
@@ -351,6 +368,59 @@ def __init__(self, default_version: str, versions: Dict[str, BlockMatrixResource
         super().__init__(default_version, versions)
 
 
+GNOMAD_PUBLIC_BUCKETS = ("gnomad-public", "gnomad-public-requester-pays")
+
+
+class GnomadPublicResource(BaseResource, ABC):
+    """Base class for the gnomAD project's public resources."""
+
+    def _get_path(self) -> str:
+        resource_source = gnomad_public_resource_configuration.source
+        if resource_source == GnomadPublicResourceSource.GNOMAD:
+            return self._path
+
+        relative_path = reduce(
```

Comment:
I might be using this wrong, but when I tested this I got something I didn't expect

```
GNOMAD_PUBLIC_BUCKETS = ("gnomad-public", "gnomad-public-requester-pays")

path2 = "gs://gnomad-public-requester-pays/release/2.1.1/ld/sv/gnomad.genomes_snv_sv.r2.1.1.afr.snv_sv.ld.ht"

relative_path = reduce(
            lambda path, bucket: path[5 + len(bucket) :]
            if path.startswith(f"gs://{bucket}")
            else path,
            GNOMAD_PUBLIC_BUCKETS,
            path2,
        )

print(relative_path)
```

I get `-requester-pays/release/2.1.1/ld/sv/gnomad.genomes_snv_sv.r2.1.1.afr.snv_sv.ld.ht` but was expecting `/release/2.1.1/ld/sv/gnomad.genomes_snv_sv.r2.1.1.afr.snv_sv.ld.ht`

Reply:
Ooh, good catch. This won't work because `gs://gnomad-public` is a prefix of `gs://gnomad-public-requester-pays`.

Reply:
Fixed in cc5d0956d0bb8c731d9fe7dc3945c605a9e36559.